src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -195,10 +194,39 @@ Device::Device(const VkInstance instance, const VkSurfaceKHR surface, bool enabl
         throw std::runtime_error("Error: vkCreateDevice failed!");
     }
 
-    // TODO: Assign internal Vulkan debug marker name to device and physical device.
+#ifndef NDEBUG
+    if (enable_vulkan_debug_markers) {
+        spdlog::debug("Initializing Vulkan debug markers.");
 
-    spdlog::debug("Initialising GPU queues.");
+        // The debug marker extension is not part of the core, so function pointers need to be loaded manually.
+        vkDebugMarkerSetObjectTag =
+            (PFN_vkDebugMarkerSetObjectTagEXT)vkGetDeviceProcAddr(device, "vkDebugMarkerSetObjectTagEXT");
```

Comment:
All of these c-style casts should really be `reinterpret_cast`

Reply:
Right.

Reply:
Done.